cvxpy/problems/problem.py

```diff
@@ -1220,7 +1221,7 @@ def _solve(self,
                 reductions = [FlipObjective()]
             else:
                 reductions = []
-            reductions = reductions + [Dnlp2Smooth()]
+            reductions = reductions + [CvxAttr2Constr(reduce_bounds=False), Dnlp2Smooth()]
```

Comment:
this allows us to get the sparsity attribute. We might also do better handling of other attributes here, but that's maybe later.